Core/clim-core/presentations/presentation-type-functions.lisp

```diff
@@ -327,67 +356,34 @@ MAYBE-SUPERTYPE, regardless of parameters."
-               (otherwise (stupid-subtypep maybe-subtype super-1)))
-             (stupid-subtypep maybe-subtype super-1))))
-      ((eq maybe-supertype-name 'or)
-       (let ((or-types (decode-parameters maybe-supertype)))
-         (some (lambda (x) (stupid-subtypep maybe-subtype x)) or-types)))
-      ((eq maybe-subtype-name 'and)
-       ;; This clause is actually not conservative, but probably in a way that
-       ;; no-one will complain about too much. Basically, we will only return
-       ;; T if the first type in the AND (which is treated specially by CLIM)
-       ;; is a subtypep of the MAYBE-SUPERTYPE.
-       (stupid-subtypep (car (decode-parameters maybe-subtype)) maybe-supertype))
-      ((and (eq maybe-subtype-name 'completion)
-            (not (eq maybe-supertype-name 'completion)))
-       (destructuring-bind (sequence
-                            &key (value-key 'identity)
-                            &allow-other-keys)
-           (decode-parameters maybe-subtype)
-         (every (lambda (elt)
-                  (presentation-typep (funcall value-key elt) maybe-supertype))
-                sequence)))
-      ((and (eq maybe-subtype-name 'subset-completion)
-            (eq maybe-supertype-name 'sequence))
-       (destructuring-bind (sequence
-                            &key (value-key 'identity)
-                            &allow-other-keys)
-           (decode-parameters maybe-subtype)
-         (let ((element-type (car (decode-parameters maybe-supertype))))
-           (every (lambda (elt)
-                    (presentation-typep (funcall value-key elt) element-type))
-                  sequence))))
-      (t
-       (let ((subtype-meta (get-ptype-metaclass maybe-subtype-name))
-             (maybe-supertype-meta (get-ptype-metaclass maybe-supertype-name)))
-         (unless (and subtype-meta maybe-supertype-meta)
-           (return-from stupid-subtypep nil))
-         (map-over-ptype-superclasses #'(lambda (super)
-                                          (when (eq maybe-supertype-meta super)
-                                            (return-from stupid-subtypep t)))
-                                      maybe-subtype-name)
-         nil)))))
+    ;; In most cases, meta types can only be compared meaningfully by
+    ;; considering the presentation type parameters, but
+    ;; STUPID-SUBTYPEP is not supposed to do that, so return the
+    ;; conservative answer.
+    ;;
+    ;; If MAYBE-SUBTYPE is a SEQUENCE type (in which case we know that
+    ;; MAYBE-SUBTYPE is not OR, AND, COMPLETION or SEQUENCE - the last
+    ;; since we first checked for MAYBE-SUBTYPE and MAYBE-SUPERTYPE
+    ;; being identical), it cannot be a subtype and we don't have to
+    ;; return true. If MAYBE-SUPERTYPE is a SEQUENCE type, the only
+    ;; remaining way, after excluding COMPLETION, MAYBE-SUBTYPE can
+    ;; have sequences as values is SUBSET-COMPLETION.
```

Comment:
I don't understand the second paragraph. I'd simply remove it (or, eventually, rewrite it in simpler manner).

Reply:
Rewritten.